Add count parameter to history endpoints

noddi opened the pull request:
PR for feature from #1266 (first comment).

drcrallen commented on indexing-service/src/main/java/io/druid/indexing/overlord/http/OverlordResource.java:
```diff
@@ -219,10 +222,21 @@ public Response setWorkerConfig(
   )
   {
     Interval theInterval = interval == null ? null : new Interval(interval);
+    if (theInterval == null && count != null && count > 0) {
```
suggest returning bad request if count is less then or equal to zero.

drcrallen commented on server/src/main/java/io/druid/server/audit/SQLAuditManager.java:
```diff
@@ -207,4 +217,82 @@ public AuditEntry map(int index, ResultSet r, StatementContext ctx)
   }
 
+  @Override
+  public List<AuditEntry> fetchAuditHistory(final String key, final String type, int limit) {
```
can these two functions be refactored a bit to not duplicate so much code?

drcrallen commented on server/src/main/java/io/druid/server/http/CoordinatorDynamicConfigsResource.java:
```diff
@@ -90,10 +91,21 @@ public Response setDynamicConfigs(final CoordinatorDynamicConfig dynamicConfig,
   )
   {
     Interval theInterval = interval == null ? null : new Interval(interval);
+    if (theInterval == null && count != null && count > 0) {
```
suggest bad request if count less than or equal to zero

drcrallen commented on server/src/main/java/io/druid/server/http/RulesResource.java:
```diff
@@ -108,10 +108,15 @@ public Response setDatasourceRules(
   )
   {
     Interval theInterval = interval == null ? null : new Interval(interval);
+    if (theInterval == null && count != null && count > 0) {
```
same thing about count

drcrallen commented on server/src/test/java/io/druid/server/audit/SQLAuditManagerTest.java:
```diff
@@ -135,6 +134,90 @@ public void testFetchAuditHistory() throws IOException
     Assert.assertEquals(entry, auditEntries.get(1));
   }
 
+  @Test
```
Can you add tests for the count <=0 case? The docs do not clarify what happens in such a case and a unit test would be helpful.

drcrallen replied:
Note this was really meant as a general comment and not really meaning "in this exact file"

drcrallen commented on server/src/test/java/io/druid/server/http/RulesResourceTest.java:
```diff
@@ -0,0 +1,135 @@
+
+    EasyMock.verify(auditManager);
+  }
+}
```
newline

noddi replied:
Done.

drcrallen commented:
@noddi This looks really good! a few comments and I think it'll be ready to go soon.

noddi commented:
@drcrallen thanks for code review!

noddi commented:
Changes:
- *fetchAuditHistory* method now throws *IllegalArgumentException* if limit is <= 0
- refactored *fetchAuditHistory* method
- removed condition *count <= 0* in http methods, now *IllegalArgumentException* is caught and HTTP 400 returned in response
- added tests in *SQLAuditManagerTest*, *RulesResourceTest* for *count <= 0*

drcrallen commented on server/src/main/java/io/druid/server/audit/SQLAuditManager.java:
```diff
@@ -207,4 +219,63 @@ public AuditEntry map(int index, ResultSet r, StatementContext ctx)
+      throws IllegalArgumentException
+  {
+    final int theLimit = getLimit(limit);
+    String queryString = "SELECT payload FROM %s WHERE type = :type";
```
Can you do the String.format right here?

noddi replied:
Done.

drcrallen commented:
Cool, please squash. I'm :+1: after squash and travis pass.

fjy commented:
@noddi Thanks for all the awesome contributions!

drcrallen commented:
Bouncing for travis

noddi commented:
@fjy Thanks!
